Add cloneContainer demo

Robert-Frampton opened the pull request.

Robert-Frampton commented:
@jbalsas, tests are green now.

jbalsas commented on demos/cloneContainer.html:
```diff
@@ -0,0 +1,90 @@
+	</div>
+	<script type="text/javascript">
+		var dragDrop = new metal.DragDrop({
+			cloneContainer: 'body',
```
Isn't this the default now? Should we have another demo for the other case?

jbalsas commented on src/Drag.js:
```diff
@@ -149,16 +149,32 @@ class Drag extends State {
+
+		if (parentNode !== this.cloneContainer) {
+			endPos.y = endPos.y - parentNode.offsetTop;
+			endPos.x = endPos.x - parentNode.offsetLeft;
```
Nit: order lines, `endPos.x` before `endPos.y`?

jbalsas commented on test/Drag.js:
```diff
@@ -552,6 +557,31 @@ describe('Drag', function() {
+
+		it('should clone drag source into "cloneContainer" element', function() {
+			drag = new Drag({
+				cloneContainer: 'body',
```
Isn't this already the default value?

jbalsas commented on test/DragDrop.js:
```diff
@@ -352,4 +356,31 @@ describe('DragDrop', function() {
+			parent.style.top = '40px';
+
+			dragDrop = new DragDrop({
+				cloneContainer: 'body',
```
Isn't this already the default value?

jbalsas commented on src/Drag.js:
```diff
@@ -719,6 +773,17 @@ Drag.STATE = {
+	cloneContainer: {
+		setter: dom.toElement,
+		validator: 'validateElementOrString_',
+		value: 'body'
```
Seeing the demos and the tests, I wonder how one could get back the old behaviour for `cloneContainer`. The way it is defined, one would need to know the `parentContainer` directly.

Maybe we could have `@type {Element|string|boolean}`, so when set to `false`, it will resort back to just appending as a sibling element? Not sure what would happen when set to `true` in that case, though 🤔

jbalsas commented:
Also, I've sent https://github.com/metal/metal-drag-drop/pull/14 with the new guidelines. Could you merge that one first and apply this on top, so we already keep it aligned?

Robert-Frampton commented:
@jbalsas, rebased and updated with the changes you requested.

I made `cloneContainer` nullable, so passing...
```
new Drag({
    cloneContainer: null
});
```

Will override the default value.

Also note that due to the `tabWidth` issue we found with `prettier-eslint` I had to do 4ac9f93112234d26baff7f7e51a2893fbbccf068 as a workaround so that CI didn't fail on the lint command.
